src/components/Modal/Modal.js

```diff
@@ -0,0 +1,169 @@
+import type { ModalOptions } from './ModalProvider';
+
+export { default as ModalProvider } from './ModalProvider';
+
+export type ScrollToProps = {
+  x?: number,
+  y: number,
+  animated: boolean,
+}
+
+type SwipeDirection = 'up' | 'down' | 'left' | 'right';
+
+// Props according to react-native-modal. The ones commented out are not forwarded
+type Props = {|
+  children?: ReactNode,
+  animationIn?: string | Object,
+  animationInTiming?: number,
+  animationOut?: string | Object,
+  animationOutTiming?: number,
+  avoidKeyboard?: boolean,
+  coverScreen?: boolean,
+  hasBackdrop?: boolean,
+  backdropColor?: string,
+  backdropOpacity?: number,
+  backdropTransitionInTiming?: number,
+  backdropTransitionOutTiming?: number,
+  customBackdrop?: ReactNode,
+  deviceHeight?: number,
+  deviceWidth?: number,
+  // onBackButtonPress?: () => void,
+  // onBackdropPress?: () => void,
+  onModalWillHide?: () => void,
+  onModalHide?: () => void,
+  onModalWillShow?: () => void,
+  onModalShow?: () => void,
+  onSwipeStart?: () => void,
+  onSwipeMove?: (percentageShown: number) => void,
+  // onSwipeComplete?: (_: { swipingDirection: SwipeDirection }) => void,
+  // onSwipeCancel?: () => void,
+  scrollOffset?: number,
+  scrollOffsetMax?: number,
+  scrollTo?: (_: ScrollToProps) => void,
+  scrollHorizontal?: boolean,
+  swipeThreshold?: number,
+  swipeDirection?: SwipeDirection | SwipeDirection[],
+  useNativeDriver?: boolean,
+  hideModalContentWhileAnimating?: boolean,
+  // propagateSwipe?: boolean,
+  style?: any,
+|};
+
+type State = {|
+  isVisible: boolean,
+|};
+
+class Modal extends React.Component<Props, State> {
+  static open(render: $PropertyType<ModalOptions, 'render'>) {
+    const instance = ModalProvider.getInstance();
+    if (instance) instance.open({ render });
+  }
+
+  static closeAll() {
+    const instance = ModalProvider.getInstance();
+    if (instance) instance.closeAll();
+  }
+
+  static contextType = ModalIdContext;
+
+  // alias for readability
+  getId = () => this.context
+
+  state = {
+    isVisible: true,
+  }
+
+  closingCallbacks: (() => void)[] = [];
+
+  callClosingCallbacks = () => {
+    this.closingCallbacks.forEach(callback => callback());
+    this.closingCallbacks = [];
+  }
+
+  componentDidMount() {
+    const instance = ModalProvider.getInstance();
+    if (instance) instance.modalInstances.set(this.getId(), this);
+  }
+
+  componentWillUnmount() {
+    // Don't let the _close method hang if RNModal's onModalHide doesn't get
+    // called before unmounting.
+    this.callClosingCallbacks();
+
+    const instance = ModalProvider.getInstance();
+    if (instance) instance.modalInstances.delete(this.getId());
+  }
+
+  close = () => {
+    const instance = ModalProvider.getInstance();
+    if (instance) instance.close(this.getId());
+  }
+
+  _close = () => {
+    this.setState({ isVisible: false });
+    return new Promise((resolve) => {
+      this.closingCallbacks.push(resolve);
+    });
+  }
```

Comment:
`close` needs to stay like it is because it's the method used externally, but I agree -- I wasn't quite happy with the naming either, so I'll try to think of something more descriptive for the internal one.

Reply:
No problem - it's not a show stopper though, just a suggestion :) I'll resolve this as it's quite minor in the grand scheme of things.